lib/client/signal_fx_client.js

```diff
@@ -48,14 +84,25 @@ SignalFxClient.prototype.API_ENDPOINT_SUFFIX = 'v1/event';
 /**
  * Send the given metrics to SignalFx.
  *
- * @param data - param object with followinf fields:
+ * @param data - param object with following fields:
  *    'cumulative_counters' (list): a list of dictionaries representing the cumulative counters to report.
  *    'gauges' (list): a list of dictionaries representing the gauges to report.
  *    'counters' (list): a list of dictionaries representing the counters to report.
  */
 SignalFxClient.prototype.send = function (data) {
   var _this = this;
+  this.rawData.push(data);
+  return this.loadAWSUniqueId
+    .then(function () {
+      _this.processingData();
```

Comment:
Since this doesn't return another promise, the following .then is unnecessary, it can be added to this block. If _this.processingData is a promise, it should be returned here.